ivadomed/loader/mri3d_subvolume_segmentation_dataset.py

```diff
@@ -75,13 +87,38 @@ def _load_filenames(self):
 
             for metadata in seg_pair[MetadataKW.INPUT_METADATA]:
                 metadata[MetadataKW.INDEX_SHAPE] = seg_pair['input'][0].shape
-            self.handlers.append((seg_pair, roi_pair))
+
+            # First time detemine cache automatically IF not specified. Otherwise, use the cache specified.
+            if self.disk_cache is None:
+                self.determine_cache_need(seg_pair, roi_pair)
+
+            if self.disk_cache:
```

Comment:
I was thinking whether we should cache 3D data _by default_ (for one of my reasons mentioned above, about having more space for the model), irrespective of the size of the dataset? Thoughts?

 I guess this also boils down to the time difference between (i) caching and retrieving vs. (ii) storing on the memory (i.e. on the fly). Seems like (ii) is going to be faster but if the (i) does not take much time, then we could cache by default!

Reply:
Sure, current system design can certainly enabling caching by default. Simply update the class default to True and we are good to go. 

The only potential downside is that all data are cached to `.ivadomed/cache` folder in the user home directory `/home/.ivadomed` or `appdata` on win, which could bloat over time without an explicit call to clear them (user will never do it, we might need to provide this function. )

Reply:
That's a good point! Maybe we can call it ourselves once the training ends? I mean there wouldn't be much use of the cached files after the training so we can clear it?